erpnext/accounts/doctype/payment_request/payment_request.py

```diff
@@ -398,6 +398,32 @@ def make_status_as_paid(doc, method):
 				doc.db_set('status', 'Paid')
 				frappe.db.commit()
 
+def update_after_payment_entry(pe, method):
+	# if payment entry covers the payment request amount, pr can be set as paid
+	# otherwise pe amount is deducted from pr grand total.
+	# in case of multiple referent sale, pr total is deducted from pe amount until all amount is attributed
+	paid_amount = pe.paid_amount
+	for ref in pe.references:
+		payment_request_name = frappe.db.get_value("Payment Request",
+			{"reference_doctype": ref.reference_doctype, "reference_name": ref.reference_name,
+				"docstatus": 1})
+
+		if payment_request_name:
+			pr = frappe.get_doc("Payment Request", payment_request_name)
+			pr_amount = pr.grand_total
+			if pr.grand_total <= paid_amount:
```

Comment:
paid_amount should not be considered based on `pe.paid_amount`. Instead it should be based on `ref.allocated_amount`.